src/stores/modalContext.tsx

```diff
@@ -0,0 +1,235 @@
+import React, { createContext, useContext, useRef, useState } from 'react'
+import { ModalRef } from '@trussworks/react-uswds'
+import { useAnalytics } from 'stores/analyticsContext'
+import { useAddBookmarkMutation } from 'operations/portal/mutations/addBookmark.g'
+import { useRemoveCollectionMutation } from 'operations/portal/mutations/removeCollection.g'
+import { useRemoveWidgetMutation } from 'operations/portal/mutations/removeWidget.g'
+import { useRemoveBookmarkMutation } from 'operations/portal/mutations/removeBookmark.g'
+import { useEditBookmarkMutation } from 'operations/portal/mutations/editBookmark.g'
+import { Widget, Bookmark as BookmarkType } from 'types/index'
+
+// Try to clean this up
+export type ModalContextType = {
+  modalId: string
+  updateModalId: (modalId: string) => void
+  modalRef: React.RefObject<ModalRef> | null | any
+  modalHeadingText: string
+  closeModal: () => void
+  onDelete: () => void
+  onSave: (url: string, label: string) => void
+  updateWidget: (widget: Widget) => void
+  updateModalText: ({
+    headingText,
+    descriptionText,
+  }: {
+    headingText: string
+    descriptionText?: string
+  }) => void
+  additionalText?: string
+  bookmark?: BookmarkType | null
+  updateBookmark: (bookmark: BookmarkType) => void
+  customLinkLabel?: string
+  updateCustomLinkLabel: (
+    customLinkLabel: string,
+    showAddWarning: boolean,
+    isAddingLink: boolean
+  ) => void
+  showAddWarning?: boolean
+  isAddingLinkContext: boolean
+}
+
+export const ModalContext = createContext<ModalContextType>({
+  modalId: '',
+  updateModalId: /* istanbul ignore next */ () => {
+    return
+  },
+  modalRef: null,
+  modalHeadingText: '',
+  closeModal: /* istanbul ignore next */ () => {
+    return
+  },
+  onDelete: /* istanbul ignore next */ () => {
+    return
+  },
+  onSave: /* istanbul ignore next */ () => {
+    return
+  },
+  updateWidget: /* istanbul ignore next */ () => {
+    return
+  },
+  updateBookmark: /* istanbul ignore next */ () => {
+    return
+  },
+  updateCustomLinkLabel: /* istanbul ignore next */ () => {
+    return
+  },
+  updateModalText: /* istanbul ignore next */ () => {
+    return
+  },
+  additionalText: '',
+  isAddingLinkContext: false,
+})
+
+export const ModalProvider = ({ children }: { children: React.ReactNode }) => {
+  const [modalHeadingText, setModalHeadingText] = useState('')
+  const [additionalText, setAdditionalText] = useState('')
+  const [modalId, setModalId] = useState('')
+  const [customLinkLabel, setCustomLinkLabel] = useState('')
+  const [showAddWarning, setShowAddWarning] = useState(false)
+  const [widgetState, setWidgetState] = useState<Widget | null>()
+  const [bookmark, setBookmark] = useState<BookmarkType | null>()
+  const [isAddingLinkContext, setIsAddingLinkContext] = useState(false)
```

Comment:
Question: I'm not sure I understand what this attribute is used for. It's name is a little unclear to me. Can you add some comments about how it's used / what it's intended to do?